### PR TITLE
Fix sync/async deadlock

### PR DESCRIPTION
**Pull request by jasper-d**
Fixes a deadlock #395 using TaskCreationOptions.RunContinuationsAsynchronously on netstandard1.6.
No integration test because I couldn't make it fail reliably (before the fix). 

Upgraded to net46 since #394 will do it anyways and it avoids an ugly workaround.

### CODE REVIEW

**Comment by watfordgnf** [comment]
I don't think its a dependency we can take, but [Microsoft's vs-threading repo](https://github.com/microsoft/vs-threading) has nice handlers for sync-over-async (aka JoinableTaskFactory) that perhaps would be a solution in this space.

**Comment by jasper-d** [comment]
I don't know vs-threading, but looks like it wouldn't work for us indeed: https://github.com/microsoft/vs-threading/blob/master/src/Microsoft.VisualStudio.Threading/Microsoft.VisualStudio.Threading.csproj#L3, more specifically https://github.com/microsoft/vs-threading/search?q=RunContinuationsAsynchronously&unscoped_q=RunContinuationsAsynchronously (that's used by the JoinableTaskFactory).

**Review by ColinSullivan1 — APPROVED** [review]
LGTM.

**Comment by ColinSullivan1** [comment]
Thanks!

File: src/NATS.Client/Conn.cs [hunk 1]
```diff
@@ -2720,24 +2720,8 @@ private Msg requestSync(string subject, byte[] data, int offset, int count, int
...
-                    {
-                        // we *should* only have one, and it should be
-                        // a NATS timeout exception.
-                        throw e;
-                    }
-
-                    throw;
-                }
+                
+                return request.Waiter.Task.GetAwaiter().GetResult();
```
Comment (watfordgnf) [inline src/NATS.Client/Conn.cs:2724]: Does this still timeout after `timeout` milliseconds?

Comment (jasper-d) [reply src/NATS.Client/Conn.cs:2724]: Yes, `timeout` is passed through `setupRequest` to `InflightRequest` where it's used to set the timeout.https://github.com/nats-io/nats.net/blob/ec0a317bde5cf0b7d88a5e63c83ae43527d0274c/src/NATS.Client/Internals/InFlightRequest.cs#L72

**Review by watfordgnf — APPROVED** [review]

**Pull request merged**